### PR TITLE
Add switch on/off api for ssh half-open detection

### PR DESCRIPTION
**Pull request by jerrytt**
Add A switch to half-open detect.

when we can predict ssh-half-open will be happen expected, we switch off detection and restore it after it happened.

- Related ticket: https://progress.opensuse.org/issues/32746
- Needles: None
- Verification run: http://10.67.132.86/tests/353#

### CODE REVIEW

**Comment by okurz** [comment]
s/action , restore/action, restore/

**Comment by mnowaksuse** [comment]
Looks like a too big brush to me. Can't you rather `stop_ssh_serial` where it's expected to be unusable (restart, shutdown, ...)?

File: backend/baseclass.pm [hunk 1]
```diff
@@ -238,6 +238,11 @@ sub run_capture_loop {
                     # Very high limits! On a working socket, the maximum hits per 10 seconds will be around 60.
                     # The maximum hits per 10 seconds saw on a half open socket was >100k
                     if (check_select_rate($buckets, $wait_time_limit, $hits_limit, fileno $fh)) {
+                        if ($self->ssh_half_open_expected) {
+                            bmwqemu::diag "expected half-open sockets in ssh";
+                            $self->stop_ssh_serial;
+                            last;
+                        }
```
Comment (okurz) [inline backend/baseclass.pm]: Hm, the indendation / complexity is IMHO too high now, too convoluted code. Also, I do not think that any "ssh" specific code should be hidden in here. How about just raising this event as a proper exception and handling it on a higher level in a specific way?

Comment (jerrytt) [reply backend/baseclass.pm]: Thanks @okurz ,
This code is in a while loop, Raising this event and handling on higher level may need more carefully .
can you give example ? thanks

**Comment by jerrytt** [comment]
> Looks like a too big brush to me. Can't you rather `stop_ssh_serial` where it's expected to be unusable (restart, shutdown, ...)?
sorry , i don't understand what you mean .
do you mean directly call stop_ssh_serial where restart/shutdown ?

**Comment by mnowaksuse** [comment]
> do you mean directly call stop_ssh_serial where restart/shutdown ?

Yes.

File: backend/baseclass.pm (hunk 1 above)
Comment (okurz) [reply backend/baseclass.pm]: Look for https://github.com/os-autoinst/os-autoinst/search?q=Exception&unscoped_q=Exception , maybe that helps

**Comment by jerrytt** [comment]
Unfortunately no, please read the issue ticket @okurz comments.

**Comment by jerrytt** [comment]
@mitiao  please take care of this issue. 
I may not have time focus on this.

**Comment by okurz** [comment]
I assume with https://github.com/os-autoinst/os-autoinst/pull/1232 we have found a better, competing approach so closing this ticket. If you feel you can provide a better, updated version feel welcome to re-open

**Pull request closed without merging**